### ISSUE TITLE
Block string parsing

### ISSUE THREAD

**Issue by abooij** [hasura/graphql-parser-hs#30]
We don't seem to support the [block strings defined in the GraphQL spec](http://spec.graphql.org/June2018/#sec-String-Value):

> For example, the following operation containing a block string:
> ```graphql
> mutation {
>   sendEmail(message: """
>     Hello,
>       World!
> 
>     Yours,
>       GraphQL.
>   """)
> }
> ```
> Is identical to the standard quoted string:
> ```graphql
> mutation {
>   sendEmail(message: "Hello,\n  World!\n\nYours,\n  GraphQL.")
> }
> ```

Note that the parsing of this is rather nontrivial:

> Since block strings represent freeform text often used in indented positions, the string value semantics of a block string excludes uniform indentation and blank initial and trailing lines via BlockStringValue().

Here, [`BlockStringValue`](http://spec.graphql.org/June2018/#BlockStringValue()) is a highly nontrivial function that basically says we should ignore any indentation that is common to all lines in a block string. Indeed, the above queries are **not** equivalent to:

```graphql
mutation {
  sendEmail(message: "    Hello,\n      World!\n\n    Yours,\n      GraphQL.\n  """)
}
```

I think the easiest solution here is to parse the string including indentation, and attempt to replicate `BlockStringValue` to remove indentation afterwards. (I say this explicitly because one can also write indentation-aware parsers using parser combinators alone.)

**Comment by lorenzo** [comment]
I was recently affected by this issue when using Hasura + remote schema permissions. I think I can work on adding support for it unless someone already is doing it.

Is it ok if I contribute the feature?

**Comment by codingkarthik** [comment]
@lorenzo It will be awesome if you can contribute the feature. I can help you with it if you need it :smile:

**Comment by vitalibarozzi** [comment]
@codingkarthik 
Hello, I'll be working in this feature for @lorenzo.
:)

**Comment by abooij** [comment]
Fixed by #46; closing.
